shared/src/main/scala/io/kaitai/struct/languages/AwkwardCompiler.scala

```diff
@@ -727,6 +726,7 @@ class AwkwardCompiler(
           s"${if (unionIndex.contains("child")) ".content<" + unionIndex.split("_").last + ">()" else ""}" + 
           s";"
         )
+        outSrc.puts(s"${nameList.last}_builder.set_fields(${nameList.last}_fields_map);")
```

Comment:
I tried moving `set_fields` into `AttrParse2`, but this actually breaks the test on the header.
While it used to parse the header as:

    array([(100, 16)], dtype=[('bank_hdrA__Zall_bank_size', '<u4'), ('bank_hdrA__Zflags', '<u4')])

now it lost the field names:

    array([(100, 16)], dtype=[('0', '<u4'), ('1', '<u4')])

here is the generated cpp
https://gist.github.com/zonca/425cc2ab0489949466b07eb1582013b2

Reply:
In Awkward Arrays, we have `bank1` as `None`
```
>>> awkward_array.to_list()
[{'index_optionA__Zbank_header': {'bank_hdrA__Zall_bank_size': 100, 'bank_hdrA__Zflags': 16}, 'index_optionA__Zbank1': None, 'index_optionA__Zbank2': {'bank32A__Zname': 'Bank', 'bank32A__Ztype': 655361, 'bank32A__Zsize': 1802395970}}]
```
Now we can see the fields of `bank1
```
>>> awkward_array.layout.form
RecordForm([RecordForm([NumpyForm('uint32'), NumpyForm('uint32')], ['bank_hdrA__Zall_bank_size', 'bank_hdrA__Zflags']), IndexedOptionForm('i64', RecordForm([ListOffsetForm('i64', NumpyForm('uint8', parameters={'__array__': 'char'}), parameters={'__array__': 'string'}), NumpyForm('uint16'), NumpyForm('uint16')], ['bank16A__Zname', 'bank16A__Ztype', 'bank16A__Zsize'])), IndexedOptionForm('i64', RecordForm([ListOffsetForm('i64', NumpyForm('uint8', parameters={'__array__': 'char'}), parameters={'__array__': 'string'}), NumpyForm('uint32'), NumpyForm('uint32')], ['bank32A__Zname', 'bank32A__Ztype', 'bank32A__Zsize']))], ['index_optionA__Zbank_header', 'index_optionA__Zbank1', 'index_optionA__Zbank2'])
```